docroot/modules/custom/foia_file/drush/foia_file.drush.inc

```diff
@@ -0,0 +1,174 @@
+      if (empty($pathPart)) {
+        $status = 'invalid';
+      }
+    }
+
+    // Log any discrepancies between filesystem and entities.
+    if ($status === 'OK' && !is_file($fileName)) {
+      \Drupal::logger('foia_file')->warning(
+        "The file @fileName was not detected on the filesystem and was marked as clean by the virus scanner.",
+        ['@fileName' => $fileName]
+        );
+    }
+    if ($status === 'Removed.' && is_file($fileName)) {
+      \Drupal::logger('foia_file')->warning(
+        "The file @fileName was detected on the filesystem and a virus was detected in it. The file should have been deleted by the scanner.",
+        ['@fileName' => $fileName]
+      );
+    }
+
+    // Determine file entity id based upon file path & name.
+    $query = \Drupal::entityQuery('file')
+      ->condition('uri', "private://webform/{$fileName}");
+    $fids = $query->execute();
+    $fid = array_values($fids)[0];
+
+    // @todo Validate that only one item in fids array.
+
+    // @todo Nice to have >> verify that submission id is same as file path.
+
+    // Update file entity based upon status.
+    if ($fid) {
+      $fileEntity = FileEntity::load($fid);
+    }
+    else {
+      $fileEntity = NULL;
+    }
+    $doNotScanStatuses = ['clean', 'virus', 'invalid'];
+    if ($fileEntity && !in_array($fileEntity->get('field_virus_scan_status')->getString(), $doNotScanStatuses)) {
+      if ($status === "OK") {
+        $fileEntity->set('field_virus_scan_status', 'clean');
+        \Drupal::logger('foia_file')->info(
+          "File @webformDir@fileName Entity ID @fid set to clean.",
+          [
+            '@webformDir' => $webformDir,
+            '@fileName' => $fileName,
+            '@fid' => $fid,
+          ]
+        );
+      }
+      elseif ($status === "Removed.") {
+        $fileEntity->set('field_virus_scan_status', 'virus');
+        \Drupal::logger('foia_file')->warning(
+          "A virus was detected in the file @webformDir@fileName associated with the Entity ID @fid. The file has been deleted.",
+          [
+            '@webformDir' => $webformDir,
+            '@fileName' => $fileName,
+            '@fid' => $fid,
+          ]
+        );
+        \Drupal::logger('foia_file')->notice(
+          "File @webformDir@fileName Entity ID @fid set to virus.",
+          [
+            '@webformDir' => $webformDir,
+            '@fileName' => $fileName,
+            '@fid' => $fid,
+          ]
+        );
+      }
+      $fileEntity->save();
+    }
+    else {
+      if ($fileEntity) {
+        \Drupal::logger('foia_file')->info(
+          "File @webformDir@fileName Entity ID @fid has already been scanned.",
+          [
+            '@webformDir' => $webformDir,
+            '@fileName' => $fileName,
+            '@fid' => $fid,
+          ]
+        );
+      }
+      else {
+        \Drupal::logger('foia_file')->warning(
+          "There is no entity associated with @fileName.",
```

Comment:
Same deal here, maybe we don't have to log this. It's nothing too spectacular. It's also not something we should really ever encounter, since files within the webform directory should always (theoretically) be associated with a file entity.